accounts-db/src/accounts_index/bucket_map_holder.rs

```diff
@@ -207,9 +258,44 @@ impl<T: IndexValue, U: DiskIndexValue + From<T> + Into<T>> BucketMapHolder<T, U>
             .and_then(|drives| drives.first())
             .map(|drive| drive.join("accounts_index_restart"));
 
-        let disk = match config.index_limit_mb {
-            IndexLimitMb::InMemOnly => None,
-            IndexLimitMb::Minimal => Some(BucketMap::new(bucket_config)),
+        let disk = match config.index_limit {
+            IndexLimit::InMemOnly => None,
+            IndexLimit::Minimal | IndexLimit::Threshold(_) => Some(BucketMap::new(bucket_config)),
+        };
+
+        // Compute threshold_entries once here
+        let threshold_entries_per_bin = match config.index_limit {
+            IndexLimit::Threshold(limit_bytes) => {
+                let size_of_uninitialized = InMemAccountsIndex::<T, U>::size_of_uninitialized(); // 32 (pubkey) + 8 (box pointer) = 40
+                let size_of_single_entry = InMemAccountsIndex::<T, U>::size_of_single_entry(); // 48 (AccountMapEntry<T>)
+
+                // Calculate bytes allocated per bin
+                let bytes_per_bin = (limit_bytes as usize) / bins;
+
+                // Calculate entries per bin accounting for load factor:
+                // At 7/8 load factor, each entry effectively uses:
+                // (7/8 * full_entry_size) + (1/8 * uninitialized_size)
+                let full_entry_size = size_of_uninitialized + size_of_single_entry;
+                let weighted_bytes_per_entry =
+                    (7 * full_entry_size + size_of_uninitialized) / 8;
+                let entries_per_bin = bytes_per_bin / weighted_bytes_per_entry;
+
+                // Calculate threshold with log2 roundup and load factor (7/8)
+                let threshold_per_bin = if entries_per_bin == 0 {
+                    0
+                } else {
+                    let entries_per_bin_log2 = entries_per_bin.ilog2();
+                    (1usize << entries_per_bin_log2) * 7 / 8
+                };
```

Comment:
How does the hashbrown hashmap work here. If I ask for capacity that is a power of 2, e.g. 16,384, will the hashmap end up allocating the next power of two (e.g. 32,768) due to needing 1/8th free for load factor?

Or will the hashmap actually allocate 16,384, but only let us use 7/8ths of it?

Reply:
> the hashmap actually allocate 16,384, but only let us use 7/8ths of it?
yes, it does so. 
therefore the effective size that we are allowed is (1usize << entries_per_bin_log2) * 7 / 8.

Reply:
I tried in rust playground. If we ask for a capacity that is power of 2, then the hashmap *will* allocate the next power of two.

https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=9ee7aae8570d2b7f81afde80af57692c